connect/transforms/src/test/java/org/apache/kafka/connect/transforms/field/FieldPathNotationTest.java

```diff
@@ -0,0 +1,149 @@
+        assertArrayEquals(new String[] {"foo", "bar.baz"}, new SingleFieldPath("foo.`bar.baz`", FieldSyntaxVersion.V2).path());
+        assertArrayEquals(new String[] {"foo.bar", "baz"}, new SingleFieldPath("`foo.bar`.baz", FieldSyntaxVersion.V2).path());
+        assertArrayEquals(new String[] {"foo", "bar", "baz"}, new SingleFieldPath("foo.`bar`.baz", FieldSyntaxVersion.V2).path());
+    }
+
+    @Test
+    void shouldBuildV2AndIgnoreBackticksThatAreNotWrapping() {
+        // Given v2 and fields including dots and backticks
+        // When backticks are wrapping a field name (i.e. withing edges or between dots)
+        // Then build a path with steps separated by dots and including non-wrapping backticks
+        assertArrayEquals(new String[] {"foo", "`bar.baz"}, new SingleFieldPath("foo.``bar.baz`", FieldSyntaxVersion.V2).path());
+        assertArrayEquals(new String[] {"foo", "bar.baz`"}, new SingleFieldPath("foo.`bar.baz``", FieldSyntaxVersion.V2).path());
+        assertArrayEquals(new String[] {"foo", "ba`r.baz"}, new SingleFieldPath("foo.`ba`r.baz`", FieldSyntaxVersion.V2).path());
+        assertArrayEquals(new String[] {"foo", "ba`r", "baz"}, new SingleFieldPath("foo.ba`r.baz", FieldSyntaxVersion.V2).path());
+        assertArrayEquals(new String[] {"foo", "`bar`", "baz"}, new SingleFieldPath("foo.``bar``.baz", FieldSyntaxVersion.V2).path());
+        assertArrayEquals(new String[] {"`foo.bar.baz`"}, new SingleFieldPath("``foo.bar.baz``", FieldSyntaxVersion.V2).path());
+    }
+
+    @Test
+    void shouldBuildV2AndEscapeBackticks() {
+        // Given v2 and fields including dots and backticks
+        // When backticks are wrapping a field name (i.e. withing edges or between dots)
+        // and wrapping backticks that are part of the field name are escaped with backslashes
+        // Then build a path with steps separated by dots and including escaped and non-wrapping backticks
+        assertArrayEquals(new String[] {"foo", "bar`.baz"}, new SingleFieldPath("foo.`bar\\`.baz`", FieldSyntaxVersion.V2).path());
+        assertArrayEquals(new String[] {"foo", "bar.`baz"}, new SingleFieldPath("foo.`bar.`baz`", FieldSyntaxVersion.V2).path());
+        assertArrayEquals(new String[] {"foo", "bar`.`baz"}, new SingleFieldPath("foo.`bar\\`.`baz`", FieldSyntaxVersion.V2).path());
+        assertArrayEquals(new String[] {"foo", "bar\\`.\\`baz"}, new SingleFieldPath("foo.`bar\\\\`.\\`baz`", FieldSyntaxVersion.V2).path());
+    }
+
+    @Test
+    void shouldFailV2WhenIncompleteBackticks() {
+        // Given v2
+        // When backticks are not closed and not escaped
+        // Then it should fail
+        ConfigException e0 = assertThrows(ConfigException.class,
+                () -> new SingleFieldPath("`foo.bar.baz", FieldSyntaxVersion.V2));
+        assertEquals(
+                "Incomplete backtick pair in path: [`foo.bar.baz], consider adding a backslash before backtick at position 0 to escape it",
```

Comment:
Should we test for the case where a backslash precedes an opening backtick?

Reply:
Good question. In the spec we have: 
> If a backtick is followed by a dot in the field name, then the backtick should be escaped with a backslash to signal that the backtick is part of the name and not closing a backtick pair.

This is missing the scenario were a backtick is included in the field name and it's at the beginning.

I think we should extend this to include "If a bactick it's at the beginning of the field name or is followed by a dot..." and account for this when building the path. wdyt?

Reply:
That makes sense to me 👍